controller/watcher.go

```diff
@@ -54,29 +54,39 @@ func newWatcher(conf *config.Config) (*hcat.Watcher, error) {
 		return nil, err
 	}
 
+	wr := watcherRetry{
+		maxRetries: maxRetries,
+		waitFunc:   retry.WaitTime,
+	}
+
 	return hcat.NewWatcher(hcat.WatcherInput{
 		Clients:         clients,
 		Cache:           hcat.NewStore(),
-		ConsulRetryFunc: retryConsul,
+		ConsulRetryFunc: wr.retryConsul,
 	}), nil
 }
 
+type watcherRetry struct {
+	maxRetries int
+	waitFunc   func(attempt uint, random *rand.Rand) int
+}
+
 // retryConsul will be used by hashicat watcher to retry polling Consul for
 // catalog changes. If polling Consul fails after retries, CTS will actually
 // exit.
 //
 // retryCount parameter is passed in by hcat. It starts at 0 (there have been
 // zero retries).
-func retryConsul(retryCount int) (bool, time.Duration) {
-	max := 8 // 8+1 retries. total wait of 8.5-12.8 min
+func (wr watcherRetry) retryConsul(retryCount int) (bool, time.Duration) {
+	max := wr.maxRetries // 8+1 retries. total wait of 8.5-12.8 min
 	logger := logging.Global().Named(hcatLogSystemName)
-	if retryCount > max {
+	if max >= 0 && retryCount > max {
 		logger.Error("error connecting with Consul even after retries", "retries", retryCount)
 		return false, 0 * time.Second
 	}
 
 	random := rand.New(rand.NewSource(time.Now().UnixNano()))
-	wait := retry.WaitTime(uint(retryCount), random)
+	wait := wr.waitFunc(uint(retryCount), random) // max wait time of 15 minutes
```

Comment:
going to remove the literal max wait time here, and instead reference the constant for future proofing